### PR TITLE
[DISCUSSION] move OS related files out of platform specific folder

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

DO NOT MERGE!

**Context: Fixes an issue?**
No. Just moves files out of the platform specific folders into src/res, libs/ or specific folders in src/.

**Does this change need documentation? What needs to be documented and how?**

**Status of this Pull Request**

 Ready for discussion what we should do with the remaining files.

**What is missing until this pull request can be merged?**
Testing and discussion. 

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [ ] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [ ] I've filled all the content above

### CODE REVIEW

File: mac/Info-xcode.plist [hunk 1]
```diff
@@ -27,7 +27,7 @@
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
 	<key>CFBundleIconFile</key>
-	<string>mainicon.icns</string>
+	<string>mainicon.icns</string> <!-- Hopefully, this should not need the path to the file since we include it via qmake -->
```
Comment (ann0see) [inline mac/Info-xcode.plist:30]: Needs to be verified.

File: Jamulus.pro [hunk 2]
```diff
@@ -1120,6 +1120,6 @@ contains(CONFIG, "disable_version_check") {
 # be sure to update .github/workflows/coding-style-check.yml and .clang-format-ignore as well.
 CLANG_FORMAT_SOURCES = $$files(*.cpp, true) $$files(*.mm, true) $$files(*.h, true)
 CLANG_FORMAT_SOURCES = $$find(CLANG_FORMAT_SOURCES, ^\(android|ios|mac|linux|src|windows\)/)
-CLANG_FORMAT_SOURCES ~= s!^\(windows/\(nsProcess|ASIOSDK2\)/|src/res/qrc_resources\.cpp\)\S*$!!g
+CLANG_FORMAT_SOURCES ~= s!^\(libs/\(NSIS|ASIOSDK2\)/|src/res/qrc_resources\.cpp\)\S*$!!g
```
Comment (ann0see) [inline Jamulus.pro:1123]: I thought that we already moved the ASIOSDK. Is that in a not-yet merged PR?

File: mac/Info-xcode.plist (hunk 1 above)
Comment (ann0see) [reply mac/Info-xcode.plist:30]: The autobuild mac version is OK - at least no images are missing as far as I can tell

**Comment by ann0see** [comment]
@hoffie you requested this PR a while ago, and I think it makes sense discussing the moving of files. I'm not yet done fully with moving everything out, but I think it's a good start?

**Comment by hoffie** [comment]
I think the direction of the PR is fine. I haven't had a look at all details yet.

By remaing files you mean the following, right?

```
$ find android/ mac/ ios/ linux/ windows/ -type f
android/res/drawable-xhdpi/icon.png
android/res/drawable-mdpi/icon.png
android/res/drawable-xxxhdpi/icon.png
android/res/drawable-hdpi/icon.png
android/res/drawable-ldpi/icon.png
android/res/drawable-xxhdpi/icon.png
android/AndroidManifest.xml
mac/Info-make-legacy.plist
mac/Info-make.plist
mac/deploy_mac.sh
mac/Jamulus.entitlements
mac/Info-xcode.plist
ios/deploy_ios.sh
ios/Info.plist
linux/deploy_deb.sh
windows/deploy_windows.ps1
windows/installer.nsi
```

I don't have a good idea for those yet. Moving all build scripts to build/ or something would still leave some stuff. Would generic platform/ directory make sense to reduce the number of top level directories at least?

**Comment by ann0see** [comment]
Yes. Some of these files can easily be moved (.entitlements etc) but the build scripts? 
I'm not sure if the windows build version creates a folder called build during compile time, so not sure...

File: Jamulus.pro [hunk 3]
```diff
@@ -322,29 +322,29 @@ SOURCES += src/sound/oboe/sound.cpp \
...
-        icons.files = distributions/jamulus.png
+        icons.files = src/res/jamulus-deb.png
 
         isEmpty(ICONSDIR_SVG) {
             ICONSDIR_SVG = share/icons/hicolor/scalable/apps/
         }
         ICONSDIR_SVG = $$absolute_path($$ICONSDIR_SVG, $$PREFIX)
         icons_svg.path = $$ICONSDIR_SVG
-        icons_svg.files = distributions/jamulus.svg distributions/jamulus-server.svg
+        icons_svg.files = src/res/jamulus-icon-2020.svg src/res/jamulus-server-icon-2020.svg
```
Comment (ann0see) [inline Jamulus.pro:340]: I don't really like the name of this file (rather remove the 2020 and icon). Other files use it probably.

File: linux/raspijamulus.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
```
Comment (ann0see) [inline linux/raspijamulus.sh:1]: This should be tested again on a new device (ping corrados if this is merged)

**Review by ann0see — COMMENTED** [review]
Another round...

**Comment by ann0see** [comment]
I think it might make sense to open separate PRs with only a few of these commits such that testing becomes easier…

**Comment by ann0see** [comment]
Ok. Now the most "dangerous" PRs are missing.

**Comment by ann0see** [comment]
Closing since https://github.com/jamulussoftware/jamulus/pull/2838 is the last one related to this PR.

**Pull request closed without merging**